src/app/test_executive/test_executive.ml

```diff
@@ -334,7 +334,10 @@ let main inputs =
           log_engine_ref := Some log_engine ;
           let event_router =
             Dsl.Event_router.create ~logger
-              ~event_reader:(Engine.Log_engine.event_reader log_engine)
+              ~event_readers:
```

Comment:
Can you combine these into a single reader?

Reply:
The log engine and the network manager both need to write into their own pipes, unless you're suggesting that we pass the same writer-pipe into both?  I thought it was vastly cleaner to have distinct pipes and to just iterate, over the readers on the reader side.